### PR TITLE
[ImageIO] Update to Xcode 9.3 Beta 1

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

File: src/imageio.cs [hunk 1]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
 		[Mac (10, 13), iOS (11,0), TV (11,0), Watch (4,0)]
 		[Field ("kCGImagePropertyFileContentsDictionary")]
 		NSString FileContentsDictionary { get; }
+		
+		[Mac (10, 9)][iOS (11,0)]
+		[Field ("kCGImagePropertyOpenEXRDictionary")]
+		NSString OpenEXRDictionary { get; }
```
Comment (spouliot) [inline src/imageio.cs]: 3 letters acronyms should be pascal cased in .net -> `Exr` (and the typo check will likely fail)

File: src/imageio.cs [hunk 2]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtAboutCvTerm")]
+		NSString IPTCExtAboutCvTerm { get; }
```
Comment (spouliot) [inline src/imageio.cs]: -> `Iptc`

File: src/imageio.cs [hunk 3]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+		[Field ("kCGImagePropertyIPTCExtDataOnScreenRegionX")]
+		NSString IPTCExtDataOnScreenRegionX { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtDataOnScreenRegionY")]
+		NSString IPTCExtDataOnScreenRegionY { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtDigitalImageGUID")]
+		NSString IPTCExtDigitalImageGUID { get; }
```
Comment (spouliot) [inline src/imageio.cs]: -> `Guid`

File: src/imageio.cs [hunk 4]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+		[Field ("kCGImagePropertyIPTCExtLinkedEncodedRightsExpr")]
+		NSString IPTCExtLinkedEncodedRightsExpr { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtLinkedEncodedRightsExprType")]
+		NSString IPTCExtLinkedEncodedRightsExprType { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtLinkedEncodedRightsExprLangID")]
+		NSString IPTCExtLinkedEncodedRightsExprLangID { get; }
```
Comment (spouliot) [inline src/imageio.cs]: -> `Id` (2 letters but not an acronym)

File: src/imageio.cs [hunk 5]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+		[Field ("kCGImagePropertyIPTCExtLocationCountryCode")]
+		NSString IPTCExtLocationCountryCode { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtLocationCountryName")]
+		NSString IPTCExtLocationCountryName { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtLocationGPSAltitude")]
+		NSString IPTCExtLocationGPSAltitude { get; }
```
Comment (spouliot) [inline src/imageio.cs]: `Gps`

File: src/imageio.cs [hunk 6]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+		[Field ("kCGImagePropertyIPTCExtArtworkCopyrightNotice")]
+		NSString IPTCExtArtworkCopyrightNotice { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtArtworkCreator")]
+		NSString IPTCExtArtworkCreator { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtArtworkCreatorID")]
+		NSString IPTCExtArtworkCreatorID { get; }
```
Comment (spouliot) [inline src/imageio.cs]: -> `Id` (not an acronym) many... Apple is also inconsistent on them

File: src/imageio.cs (hunk 1 above)
Comment (chamons) [reply src/imageio.cs]: Agreed

File: src/imageio.cs (hunk 2 above)
Comment (chamons) [reply src/imageio.cs]: We have a _ton_ of IPTCs already and I was trying to stay consistent - https://github.com/xamarin/xamarin-macios/blob/master/src/imageio.cs#L52

Comment (spouliot) [reply src/imageio.cs]: make sense, it looked like a new one :)

File: src/imageio.cs [hunk 7]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+		[Field ("kCGImagePropertyIPTCExtArtworkSource")]
+		NSString IPTCExtArtworkSource { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtArtworkSourceInventoryNo")]
+		NSString IPTCExtArtworkSourceInventoryNo { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtArtworkSourceInvURL")]
+		NSString IPTCExtArtworkSourceInvURL { get; }
```
Comment (spouliot) [inline src/imageio.cs]: URL -> Url

File: src/imageio.cs [hunk 8]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+		[Field ("kCGImagePropertyIPTCExtLocationCountryName")]
+		NSString IPTCExtLocationCountryName { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtLocationGPSAltitude")]
+		NSString IPTCExtLocationGpsAltitude { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtLocationGPSLatitude")]
+		NSString IPTCExtLocationGPSLatitude { get; }
```
Comment (spouliot) [inline src/imageio.cs]: GPS -> Gps

File: src/imageio.cs [hunk 9]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtLocationGPSLongitude")]
+		NSString IPTCExtLocationGPSLongitude { get; }
```
Comment (spouliot) [inline src/imageio.cs]: GSP -> Gps

File: src/imageio.cs [hunk 10]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+		[Field ("kCGImagePropertyIPTCExtProductInImage")]
+		NSString IPTCExtProductInImage { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtProductInImageDescription")]
+		NSString IPTCExtProductInImageDescription { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtProductInImageGTIN")]
+		NSString IPTCExtProductInImageGTIN { get; }
```
Comment (spouliot) [inline src/imageio.cs]: GTIN -> Gtin (likely, unless it's two, two letters acronyms)

File: src/imageio.cs [hunk 11]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+		[Field ("kCGImagePropertyIPTCExtRatingRegionCountryCode")]
+		NSString IPTCExtRatingRegionCountryCode { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtRatingRegionCountryName")]
+		NSString IPTCExtRatingRegionCountryName { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtRatingRegionGPSAltitude")]
+		NSString IPTCExtRatingRegionGPSAltitude { get; }
```
Comment (spouliot) [inline src/imageio.cs]: GPS -> Gps

File: src/imageio.cs [hunk 12]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtRatingRegionGPSLatitude")]
+		NSString IPTCExtRatingRegionGPSLatitude { get; }
```
Comment (spouliot) [inline src/imageio.cs]: GPS -> Gps

File: src/imageio.cs [hunk 13]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtRatingRegionGPSLongitude")]
+		NSString IPTCExtRatingRegionGPSLongitude { get; }
```
Comment (spouliot) [inline src/imageio.cs]: GPS -> Gps

File: src/imageio.cs [hunk 14]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+		[Field ("kCGImagePropertyIPTCExtRatingValue")]
+		NSString IPTCExtRatingValue { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtRatingValueLogoLink")]
+		NSString IPTCExtRatingValueLogoLink { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtRegistryID")]
+		NSString IPTCExtRegistryID { get; }
```
Comment (spouliot) [inline src/imageio.cs]: ID -> Id

File: src/imageio.cs [hunk 15]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtRegistryEntryRole")]
+		NSString IPTCExtRegistryEntryRole { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtRegistryItemID")]
+		NSString IPTCExtRegistryItemID { get; }
```
Comment (spouliot) [inline src/imageio.cs]: ID -> Id

File: src/imageio.cs [hunk 16]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
...
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
+		[Field ("kCGImagePropertyIPTCExtRegistryOrganisationID")]
+		NSString IPTCExtRegistryOrganisationID { get; }
```
Comment (spouliot) [inline src/imageio.cs]: ID -> Id

File: src/imageio.cs [hunk 17]
```diff
@@ -901,6 +901,734 @@ interface CGImageProperties {
 		[Mac (10, 13), iOS (11,0), TV (11,0), Watch (4,0)]
 		[Field ("kCGImagePropertyFileContentsDictionary")]
 		NSString FileContentsDictionary { get; }
+		
+		[Mac (10, 9)][iOS (11,0)]
+		[Field ("kCGImagePropertyOpenEXRDictionary")]
+		NSString OpenExrDictionary { get; }
+
+		[Mac (10, 13, 4)][iOS (11, 3)]
```
Comment (dalexsoto) [inline src/imageio.cs]: do we need `onlyOn64: true` here? and on the ones below?

Comment (spouliot) [reply src/imageio.cs]: it's an old framework so there's a good chance the constants are in 32 bits only
intro should spot such cases

**Comment by spouliot** [comment]
xtro sanity checks failed -> https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/6154/Test_Report/

**Comment by chamons** [comment]
🤔  this passes locally..
```
$ make wrench
MSBuild auto-detection: using msbuild version '15.0' from '/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/msbuild/15.0/bin/'.
All packages listed in packages.config are already installed.
BUILD    build
rm -f *.unclassified
mono64 --debug bin/Debug/xtro-sharpie.exe iphoneos11.3-arm64.pch ../../_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/64bits/Xamarin.iOS.dll ../../_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/OpenTK-1.0.dll
mono64 --debug bin/Debug/xtro-sharpie.exe watchos4.3-armv7.pch ../../_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/32bits/Xamarin.WatchOS.dll
mono64 --debug bin/Debug/xtro-sharpie.exe appletvos11.3-arm64.pch ../../_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/mono/Xamarin.TVOS/Xamarin.TVOS.dll ../../_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/mono/Xamarin.TVOS/OpenTK-1.0.dll
mono64 --debug bin/Debug/xtro-sharpie.exe macosx10.13-x86_64.pch ../../_mac-build/Library/Frameworks/Xamarin.Mac.framework/Versions/git/lib/x86_64/mobile/Xamarin.Mac.dll
mono64 --debug xtro-sanity/bin/Debug/xtro-sanity.exe .
Sanity check passed
rm -f *.raw
```

**Comment by chamons** [comment]
@spouliot 👍 ?

**Comment by chamons** [comment]
Bah. I assumed incorrectly that I didn't need to run the typo tests. Fixing...

**Comment by spouliot** [comment]
@chamons the `Gtin` was added inside a `#if MONOMAC` so it fails on iOS and tvOS

**Pull request merged**